### ISSUE TITLE
undefined method `exitstatus' for nil:NilClass (NoMethodError)

### ISSUE THREAD

**Issue by samsonpaul87** [fastlane/fastlane#5928]
## New Issue Checklist

- [ ] Updated fastlane to the latest version
- [ ] I have read the [Contribution Guidelines](https://github.com/fastlane/fastlane/blob/master/CONTRIBUTING.md)

### Issue Description
getting undefined method `exitstatus' for nil:NilClass (NoMethodError)

##### Complete output when running fastlane, including the stack trace and command used
```
root@me-ThinkPad-L412:/home/me/MyProjects/Mi_Android_Studio_Project# fastlane alpha
[10:05:57]: -------------------------------------------------
[10:05:57]: --- Step: Verifying required fastlane version ---
[10:05:57]: -------------------------------------------------
[10:05:57]: fastlane version valid
[10:05:57]: ------------------------------
[10:05:57]: --- Step: default_platform ---
[10:05:57]: ------------------------------
[10:05:57]: Driving the lane 'android alpha' 🚀
[10:05:57]: ------------------------
[10:05:57]: --- Step: screengrab ---
[10:05:57]: ------------------------
[10:05:57]: The `aapt` command could not be found on your PATH
[10:05:57]: Please ensure that the Android SDK is installed and you have the build tools downloaded and present on your PATH

+-----------------------------+------------------------------------------------+
|                         Summary for screengrab 0.3.1                         |
+-----------------------------+------------------------------------------------+
| locales                     | ["en-US"]                                      |
| clear_previous_screenshots  | true                                           |
| app_package_name            | com.miproject.helloMe                   |
| app_apk_path                | /home/me/MyProjects/helloMe.apk                |
| tests_package_name          | com.miproject.helloMe.test              |
| output_directory            | fastlane/metadata/android                      |
| skip_open_summary           | true                                           |
| test_instrumentation_runner | android.support.test.runner.AndroidJUnitRunner |
| ending_locale               | en-US                                          |
| device_type                 | phone                                          |
+-----------------------------+------------------------------------------------+

[10:05:58]: Limiting the test classes run by `screengrab` to just those that generate screenshots can make runs faster.
[10:05:58]: Consider using the :use_tests_in_classes or :use_tests_in_packages option, and organize your tests accordingly.
[10:05:58]: Clearing phoneScreenshots within fastlane/metadata/android
[10:05:58]: $ adb devices -l
[10:05:58]: ▸ adb: /home/me/rubystack-2.0.0-39/common/lib/libcrypto.so.1.0.0: no version information available (required by adb)
[10:05:58]: ▸ List of devices attached
[10:05:58]: ▸ G4AXGV2448524HB        device usb:1-1.2 product:WW_Phone model:ASUS_X014D device:ASUS_X014D_1
[10:05:58]: Variable Dump:
[10:05:58]: {:DEFAULT_PLATFORM=>:android, :PLATFORM_NAME=>:android, :LANE_NAME=>"android alpha"}
[10:05:58]: undefined method `exitstatus' for nil:NilClass

+------+-------------------------------------+-------------+
|                     fastlane summary                     |
+------+-------------------------------------+-------------+
| Step | Action                              | Time (in s) |
+------+-------------------------------------+-------------+
| 1    | Verifying required fastlane version | 0           |
| 2    | default_platform                    | 0           |
| 3    | screengrab                          | 0           |
+------+-------------------------------------+-------------+

[10:05:58]: fastlane finished with errors
Successfully submitted a crash report. If this is a problem with one of the tools specifically,
please submit an issue on GitHub and attach the following number to it: 'aeb813a42d1de61fab2991aaca67c3b0'
The crash report has been stored locally '/tmp/sentry_aeb813a42d1de61fab2991aaca67c3b0.json'
/home/me/rubystack-2.0.0-39/ruby/lib/ruby/gems/2.0.0/gems/fastlane_core-0.41.2/lib/fastlane_core/command_executor.rb:79:in `execute': [!] undefined method `exitstatus' for nil:NilClass (NoMethodError)
	from /home/me/rubystack-2.0.0-39/ruby/lib/ruby/gems/2.0.0/gems/screengrab-0.3.1/lib/screengrab/runner.rb:80:in `select_device'
	from /home/me/rubystack-2.0.0-39/ruby/lib/ruby/gems/2.0.0/gems/screengrab-0.3.1/lib/screengrab/runner.rb:52:in `run'
	from /home/me/rubystack-2.0.0-39/ruby/lib/ruby/gems/2.0.0/gems/fastlane-1.80.0/lib/fastlane/actions/screengrab.rb:18:in `run'
	from /home/me/rubystack-2.0.0-39/ruby/lib/ruby/gems/2.0.0/gems/fastlane-1.80.0/lib/fastlane/runner.rb:143:in `block (2 levels) in execute_action'
	from /home/me/rubystack-2.0.0-39/ruby/lib/ruby/gems/2.0.0/gems/fastlane-1.80.0/lib/fastlane/actions/actions_helper.rb:35:in `execute_action'
	from /home/me/rubystack-2.0.0-39/ruby/lib/ruby/gems/2.0.0/gems/fastlane-1.80.0/lib/fastlane/runner.rb:129:in `block in execute_action'
	from /home/me/rubystack-2.0.0-39/ruby/lib/ruby/gems/2.0.0/gems/fastlane-1.80.0/lib/fastlane/runner.rb:128:in `chdir'
	from /home/me/rubystack-2.0.0-39/ruby/lib/ruby/gems/2.0.0/gems/fastlane-1.80.0/lib/fastlane/runner.rb:128:in `execute_action'
	from /home/me/rubystack-2.0.0-39/ruby/lib/ruby/gems/2.0.0/gems/fastlane-1.80.0/lib/fastlane/fast_file.rb:149:in `method_missing'
	from Fastfile:40:in `block (2 levels) in parsing_binding'
	from /home/me/rubystack-2.0.0-39/ruby/lib/ruby/gems/2.0.0/gems/fastlane-1.80.0/lib/fastlane/lane.rb:36:in `call'
	from /home/me/rubystack-2.0.0-39/ruby/lib/ruby/gems/2.0.0/gems/fastlane-1.80.0/lib/fastlane/lane.rb:36:in `call'
	from /home/me/rubystack-2.0.0-39/ruby/lib/ruby/gems/2.0.0/gems/fastlane-1.80.0/lib/fastlane/runner.rb:52:in `block in execute'
	from /home/me/rubystack-2.0.0-39/ruby/lib/ruby/gems/2.0.0/gems/fastlane-1.80.0/lib/fastlane/runner.rb:46:in `chdir'
	from /home/me/rubystack-2.0.0-39/ruby/lib/ruby/gems/2.0.0/gems/fastlane-1.80.0/lib/fastlane/runner.rb:46:in `execute'
	from /home/me/rubystack-2.0.0-39/ruby/lib/ruby/gems/2.0.0/gems/fastlane-1.80.0/lib/fastlane/lane_manager.rb:46:in `cruise_lane'
	from /home/me/rubystack-2.0.0-39/ruby/lib/ruby/gems/2.0.0/gems/fastlane-1.80.0/lib/fastlane/command_line_handler.rb:30:in `handle'
	from /home/me/rubystack-2.0.0-39/ruby/lib/ruby/gems/2.0.0/gems/fastlane-1.80.0/bin/fastlane:38:in `block (2 levels) in run'
	from /home/me/rubystack-2.0.0-39/ruby/lib/ruby/gems/2.0.0/gems/commander-4.3.5/lib/commander/command.rb:178:in `call'
	from /home/me/rubystack-2.0.0-39/ruby/lib/ruby/gems/2.0.0/gems/commander-4.3.5/lib/commander/command.rb:178:in `call'
	from /home/me/rubystack-2.0.0-39/ruby/lib/ruby/gems/2.0.0/gems/commander-4.3.5/lib/commander/command.rb:153:in `run'
	from /home/me/rubystack-2.0.0-39/ruby/lib/ruby/gems/2.0.0/gems/commander-4.3.5/lib/commander/runner.rb:428:in `run_active_command'
	from /home/me/rubystack-2.0.0-39/ruby/lib/ruby/gems/2.0.0/gems/fastlane_core-0.41.2/lib/fastlane_core/ui/fastlane_runner.rb:23:in `run!'
	from /home/me/rubystack-2.0.0-39/ruby/lib/ruby/gems/2.0.0/gems/commander-4.3.5/lib/commander/delegates.rb:15:in `run!'
	from /home/me/rubystack-2.0.0-39/ruby/lib/ruby/gems/2.0.0/gems/fastlane-1.80.0/bin/fastlane:158:in `run'
	from /home/me/rubystack-2.0.0-39/ruby/lib/ruby/gems/2.0.0/gems/fastlane-1.80.0/bin/fastlane:164:in `<top (required)>'
	from /home/me/rubystack-2.0.0-39/ruby/bin/fastlane:22:in `load'
	from /home/me/rubystack-2.0.0-39/ruby/bin/fastlane:22:in `<main>'

```

### Configuration Files

Please copy the complete content of your `Fastfile` and any other configuration files you use below:

**Fastfile:**
------------------
```ruby
default_platform :android

platform :android do

  desc "Alpha release app."
  lane :alpha do |options|

    screengrab(
   locales: ['en-US'],
   clear_previous_screenshots: 'true',
   app_package_name: "com.miproject.helloMe",
   app_apk_path: "/home/me/MyProjects/helloMe.apk"

 )
       
end
```

```ruby
# INSERT CONTENT HERE
```

### Environment

fastlane version (run `fastlane -v`): fastlane 1.80.0

Do you use bundler to execute fastlane (i.e. `bundle exec fastlane`)?
No

Do you use a Ruby environment manager (e.g. `chruby`, `rbenv`, `rvm`)?
No

**Comment by nafu** [comment]
@samsonpaul87 

Could you ensure this from error messages?

> 10:05:57: The aapt command could not be found on your PATH
> 10:05:57: Please ensure that the Android SDK is installed and you have the build tools downloaded and present on your PATH

**Comment by asfalcone** [comment]
@samsonpaul87 Per the new issue checklist, please update to the latest version of `fastlane` and `screengrab` so that we can accurately diagnose the problem you're running into. Thanks 👍

**Comment by samsonpaul87** [comment]
ok. I will update and let you know. :)

**Comment by samsonpaul87** [comment]
@asfalcone .. can you please take a look into the following. Please help me what I missed.

```
root@me-ThinkPad-L412:/home/me/MyProjects/Migym_Android_Studio# sudo gem update fastlane
Updating installed gems
Nothing to update
root@me-ThinkPad-L412:/home/me/MyProjects/Migym_Android_Studio# sudo gem update screengrab
Updating installed gems
Nothing to update
root@me-ThinkPad-L412:/home/me/MyProjects/Migym_Android_Studio# fastlane -v
fastlane 1.80.0
root@me-ThinkPad-L412:/home/me/MyProjects/Migym_Android_Studio# screengrab -v
screengrab 0.3.1
```

**Comment by mfurtak** [comment]
@samsonpaul87 What is the output if you run `sudo gem which fastlane` and `sudo gem which screengrab`?

**Comment by samsonpaul87** [comment]
@mfurtak .. I am getting following output
```

root@me-ThinkPad-L412:~/rubystack-2.0.0-39# sudo gem which fastlane
/home/me/rubystack-2.0.0-39/ruby/lib/ruby/gems/2.0.0/gems/fastlane-1.80.0/lib/fastlane.rb
root@me-ThinkPad-L412:~/rubystack-2.0.0-39# sudo gem which screengrab
/home/me/rubystack-2.0.0-39/ruby/lib/ruby/gems/2.0.0/gems/screengrab-0.3.1/lib/screengrab.rb
```

**Comment by mfurtak** [comment]
Thanks! That is very strange.

I'm guessing based on your path... are you using [Ruby Stack](https://bitnami.com/stack/ruby)? If so, I have to wonder whether that's doing something strange...

If you do:

```
sudo gem uninstall screengrab

sudo gem install screengrab
```

what is the result? What version is installed? The latest version available on RubyGems is [0.5.1](https://rubygems.org/gems/screengrab)

**Comment by samsonpaul87** [comment]
@mfurtak .. I am getting the following output as per your suggestion. Please have a look. :)


```
root@me-ThinkPad-L412:/home/me/MyProjects# sudo gem uninstall screengrab
INFO:  gem "screengrab" is not installed
root@me-ThinkPad-L412:/home/me/MyProjects# gem uninstall screengrab

You have requested to uninstall the gem:
	screengrab-0.3.1

fastlane-1.80.0 depends on screengrab (< 1.0.0, >= 0.3.1)
If you remove this gem, these dependencies will not be met.
Continue with Uninstall? [yN]  y
Remove executables:
	screengrab

in addition to the gem? [Yn]  y
Removing screengrab
Successfully uninstalled screengrab-0.3.1
root@me-ThinkPad-L412:/home/me/MyProjects# gem install screengrab
Fetching: commander-4.4.0.gem (100%)
Successfully installed commander-4.4.0
Fetching: gh_inspector-1.0.2.gem (100%)
Successfully installed gh_inspector-1.0.2
Fetching: credentials_manager-0.16.0.gem (100%)
Successfully installed credentials_manager-0.16.0
Fetching: fastlane_core-0.50.3.gem (100%)
Successfully installed fastlane_core-0.50.3
Fetching: screengrab-0.5.1.gem (100%)
Successfully installed screengrab-0.5.1
Parsing documentation for commander-4.4.0
Installing ri documentation for commander-4.4.0
Parsing documentation for gh_inspector-1.0.2
Installing ri documentation for gh_inspector-1.0.2
Parsing documentation for credentials_manager-0.16.0
Installing ri documentation for credentials_manager-0.16.0
Parsing documentation for fastlane_core-0.50.3
Installing ri documentation for fastlane_core-0.50.3
Parsing documentation for screengrab-0.5.1
Installing ri documentation for screengrab-0.5.1
5 gems installed
root@me-ThinkPad-L412:/home/me/MyProjects# screengrab -v
screengrab 0.5.1
```

**Comment by mfurtak** [comment]
OK, so just to tie a bow on this secondary mystery, there seems to be a difference between the gem sets that are being managed by running `gem` vs. `sudo gem` on your machine!

It's great that you got `screengrab` upgraded. 👍  If you now `gem update fastlane` (no `sudo`) does that get onto the latest version as well?

After that, can you confirm that the error is still present? Thanks for your help & patience! 🚀

**Comment by samsonpaul87** [comment]
Hi @mfurtak , Can you please check my current output, on running screengrab. Thanks :)


```
root@me-ThinkPad-L412:/home/me/MyProjects/Mi_Android_Studio_Project# bundle exec fastlane alpha
[10:57:53]: -------------------------------------------------
[10:57:53]: --- Step: Verifying required fastlane version ---
[10:57:53]: -------------------------------------------------
[10:57:53]: fastlane version valid
[10:57:53]: ------------------------------
[10:57:53]: --- Step: default_platform ---
[10:57:53]: ------------------------------
[10:57:54]: Driving the lane 'android alpha' 🚀
[10:57:54]: ------------------------
[10:57:54]: --- Step: screengrab ---
[10:57:54]: ------------------------
[10:57:54]: The `aapt` command could not be found on your PATH
[10:57:54]: Please ensure that the Android SDK is installed and you have the build tools downloaded and present on your PATH

+-----------------------------+--------------------------------------------------------------------+
|                                   Summary for screengrab 0.5.1                                   |
+-----------------------------+--------------------------------------------------------------------+
| locales                     | ["en-US"]                                                          |
| clear_previous_screenshots  | true                                                               |
| app_package_name            | com.miproject.helloMe                                       |
| app_apk_path                | /home/me/MyProjects/helloMe.apk					   |
| tests_package_name          | com.miproject.helloMe.test                                  |
| output_directory            | fastlane/metadata/android                                          |
| skip_open_summary           | true                                                               |
| test_instrumentation_runner | android.support.test.runner.AndroidJUnitRunner                     |
| ending_locale               | en-US                                                              |
| device_type                 | phone                                                              |
+-----------------------------+--------------------------------------------------------------------+

[10:57:54]: Limiting the test classes run by `screengrab` to just those that generate screenshots can make runs faster.
[10:57:54]: Consider using the :use_tests_in_classes or :use_tests_in_packages option, and organize your tests accordingly.
[10:57:54]: Clearing phoneScreenshots within fastlane/metadata/android
[10:57:54]: $ adb devices -l
[10:57:54]: ▸ adb: /home/me/rubystack-2.0.0-39/common/lib/libcrypto.so.1.0.0: no version information available (required by adb)
[10:57:54]: ▸ List of devices attached
[10:57:54]: ▸ G4AXGV2448524HB        device usb:1-1.2 product:WW_Phone model:ASUS_X014D device:ASUS_X014D_1
[10:57:54]: Multiple connected devices, selecting the first one
[10:57:54]: To specify which connected device to use, use the -s (specific_device) config option
[10:57:54]: $ adb -s adb: shell echo \$EXTERNAL_STORAGE
[10:57:54]: ▸ adb: /home/me/rubystack-2.0.0-39/common/lib/libcrypto.so.1.0.0: no version information available (required by adb)
[10:57:54]: ▸ error: device not found
adb: /home/me/rubystack-2.0.0-39/common/lib/libcrypto.so.1.0.0: no version information available (required by adb)
error: device not found
[10:57:54]: Exit status: 255
[10:57:54]: Variable Dump:
[10:57:54]: {:DEFAULT_PLATFORM=>:android, :PLATFORM_NAME=>:android, :LANE_NAME=>"android alpha"}
[10:57:54]: Exit status: 255

+------+-------------------------------------+-------------+
|                     fastlane summary                     |
+------+-------------------------------------+-------------+
| Step | Action                              | Time (in s) |
+------+-------------------------------------+-------------+
| 1    | Verifying required fastlane version | 0           |
| 2    | default_platform                    | 0           |
| 3    | screengrab                          | 0           |
+------+-------------------------------------+-------------+

[10:57:54]: fastlane finished with errors

[!] Exit status: 255
```

**Comment by mfurtak** [comment]
Got it. Thanks! I think I see the problem.

Your `adb` command is producing a warning ("adb: /home/me/rubystack-2.3.1-4/common/lib/libcrypto.so.1.0.0 ...") as part of its output that is causing _screengrab_ to fail.

I will see if I can fix `screengrab` to be more resilient to this 👍

**Comment by mfurtak** [comment]
If you want to try to work around this problem in the meantime, you can try passing `-s e1dbf228` to your `screengrab` command to manually select your device.

**Comment by mfurtak** [comment]
Hi @samsonpaul87 - I've just released [`screengrab` 0.5.2](https://github.com/fastlane/fastlane/releases/tag/screengrab/0.5.2) with some changes to  hopefully make this better for you. 🚀 
 
Can you upgrade your `screengrab` gem once more and let me know how it looks? Thanks! 🙌

**Comment by samsonpaul87** [comment]
@mfurtak ... See my output with  screengrab 0.5.2 . Please help me with this.

```

root@me-ThinkPad-L412:~/MyProjects/Mi_Android_Studio_Project# screengrab
[23:58:38]: To not be asked about this value, you can specify it using 'app_package_name'
The package name of the app under test (e.g. com.yourcompany.yourapp): com.miproject.helloMe-free
[23:59:10]: The `aapt` command could not be found on your PATH
[23:59:10]: Please ensure that the Android SDK is installed and you have the build tools downloaded and present on your PATH

+-----------------------------+------------------------------------------------+
|                         Summary for screengrab 0.5.2                         |
+-----------------------------+------------------------------------------------+
| app_package_name            | com.miproject.helloMe-free                          |
| tests_package_name          | com.miproject.helloMe-free.test   		       |
| locales                     | ["en-US"]                                      |
| clear_previous_screenshots  | false                                          |
| output_directory            | fastlane/metadata/android                      |
| skip_open_summary           | true                                           |
| test_instrumentation_runner | android.support.test.runner.AndroidJUnitRunner |
| ending_locale               | en-US                                          |
| device_type                 | phone                                          |
+-----------------------------+------------------------------------------------+

[23:59:11]: Limiting the test classes run by `screengrab` to just those that generate screenshots can make runs faster.
[23:59:11]: Consider using the :use_tests_in_classes or :use_tests_in_packages option, and organize your tests accordingly.
[23:59:11]: $ adb devices -l
[23:59:11]: ▸ adb: /home/me/rubystack-2.3.1-4/common/lib/libcrypto.so.1.0.0: no version information available (required by adb)
[23:59:11]: ▸ List of devices attached
[23:59:11]: ▸ G4AXGV2448524HB        device usb:1-1.1 product:WW_Phone model:ASUS_X014D device:ASUS_X014D_1
[23:59:11]: $ adb -s G4AXGV2448524HB shell echo \$EXTERNAL_STORAGE
[23:59:11]: ▸ adb: /home/me/rubystack-2.3.1-4/common/lib/libcrypto.so.1.0.0: no version information available (required by adb)
[23:59:11]: ▸ /storage/emulated/legacy
[23:59:11]: Cleaning screenshots on device
[23:59:11]: To not be asked about this value, you can specify it using 'app_apk_path'
[23:59:11]: Select your debug app APK
1. android/build/outputs/apk/android-helloMe-paid-release-unaligned.apk
2. android/build/outputs/apk/android-helloMe-free-release-unaligned.apk
?  2
[23:59:14]: To not be asked about this value, you can specify it using 'tests_apk_path'
[23:59:14]: Select your debug tests APK
1. android/build/outputs/apk/android-helloMe-paid-release-unaligned.apk
2. android/build/outputs/apk/android-helloMe-free-release-unaligned.apk
?  2
[23:59:16]: The `aapt` command could not be found on your system, so your app APK could not be validated
[23:59:16]: Installing app APK
[23:59:16]: $ adb -s G4AXGV2448524HB install -r android/build/outputs/apk/android-helloMe-free-release-unaligned.apk
[23:59:16]: ▸ adb: /home/me/rubystack-2.3.1-4/common/lib/libcrypto.so.1.0.0: no version information available (required by adb)
[23:59:21]: ▸ 4112 KB/s (18977626 bytes in 4.506s)
[23:59:22]: ▸ pkg: /data/local/tmp/android-helloMe-free-release-unaligned.apk
[23:59:42]: ▸ Success
[23:59:42]: Installing tests APK
[23:59:42]: $ adb -s G4AXGV2448524HB install -r android/build/outputs/apk/android-helloMe-free-release-unaligned.apk
[23:59:42]: ▸ adb: /home/me/rubystack-2.3.1-4/common/lib/libcrypto.so.1.0.0: no version information available (required by adb)
[23:59:47]: ▸ 4089 KB/s (18977626 bytes in 4.531s)
[23:59:47]: ▸ pkg: /data/local/tmp/android-helloMe-free-release-unaligned.apk
[00:00:10]: ▸ Success
[00:00:10]: Granting the permission necessary to change locales on the device
[00:00:10]: $ adb -s G4AXGV2448524HB shell pm grant com.miproject.helloMe-free android.permission.CHANGE_CONFIGURATION
[00:00:10]: ▸ adb: /home/me/rubystack-2.3.1-4/common/lib/libcrypto.so.1.0.0: no version information available (required by adb)
[00:00:11]: ▸ Operation not allowed: java.lang.SecurityException: Package com.miproject.helloMe-free has not requested permission android.permission.CHANGE_CONFIGURATION
[00:00:11]: $ adb -s G4AXGV2448524HB shell getprop ro.build.version.sdk
[00:00:11]: ▸ adb: /home/me/rubystack-2.3.1-4/common/lib/libcrypto.so.1.0.0: no version information available (required by adb)
[00:00:11]: ▸ 22
[00:00:11]: Running tests for locale: en-US
[00:00:11]: $ adb -s G4AXGV2448524HB shell am instrument --no-window-animation -w \
-e testLocale en_US \
-e endingLocale en_US \
com.miproject.helloMe-free.test/android.support.test.runner.AndroidJUnitRunner
[00:00:11]: ▸ adb: /home/me/rubystack-2.3.1-4/common/lib/libcrypto.so.1.0.0: no version information available (required by adb)
[00:00:11]: ▸ INSTRUMENTATION_STATUS: id=ActivityManagerService
[00:00:11]: ▸ INSTRUMENTATION_STATUS: Error=Unable to find instrumentation info for: ComponentInfo{com.miproject.helloMe-free.test/android.support.test.runner.AndroidJUnitRunner}
[00:00:11]: ▸ INSTRUMENTATION_STATUS_CODE: -1
[00:00:11]: ▸ android.util.AndroidException: INSTRUMENTATION_FAILED: com.miproject.helloMe-free.test/android.support.test.runner.AndroidJUnitRunner
[00:00:11]: ▸ at com.android.commands.am.Am.runInstrument(Am.java:977)
[00:00:11]: ▸ at com.android.commands.am.Am.onRun(Am.java:317)
[00:00:11]: ▸ at com.android.internal.os.BaseCommand.run(BaseCommand.java:47)
[00:00:11]: ▸ at com.android.commands.am.Am.main(Am.java:97)
[00:00:11]: ▸ at com.android.internal.os.RuntimeInit.nativeFinishInit(Native Method)
[00:00:11]: ▸ at com.android.internal.os.RuntimeInit.main(RuntimeInit.java:249)
[00:00:11]: Pulling captured screenshots from the device
[00:00:11]: Make sure you've used Screengrab.screenshot() in your tests and that your expected tests are being run.

[!] No screenshots were detected 📷❌

```

**Comment by mfurtak** [comment]
Hi @samsonpaul87 

For run 1): Can you share your `build.gradle` configuration? Are you specifying a different test runner than `android.support.test.runner.AndroidJUnitRunner`? If so, you will need to tell _screengrab_ about that class using the `test_instrumentation_runner` option, either on the command line 

```
screengrab --test_instrumentation_runner com.my.app.MyJUnitRunner
```

or in your `Fastfile` as a parameter to the `screengrab` action

```ruby
screengrab(test_instrumentation_runner: 'com.my.app.MyJUnitRunner')
```

or in your `fastlane/Screengrabfile`

```ruby
test_instrumentation_runner 'com.my.app.MyJUnitRunner'
```

Run 2) looks like a problem with ADB unrelated to _screengrab_. It may be useful to make sure that ADB is running and accepting connections by running `adb devices` before trying to run _screengrab_ 👍

**Comment by samsonpaul87** [comment]
@mfurtak ... Sorry for the late reply. I was on a vacation. 

I just got it working on my sample project :) .

I did the following

- upgraded screengrab to latest version
- added missing permissions in debug manifest file
- updated my gradle version

That's it :+1: 

Till now, I was working on only screengrab file. Right now, I am trying to move it to fastfile so that I can take screenshots and upload the apk (using Supply) with a single fastlane command. I will come up with more queries later.

Thank you for your help and support @mfurtak . You helped me a lot.
